### PR TITLE
Make Activity Serializable on NET 4.5

### PR DESCRIPTION
**Pull request by lmolkova**
In order to store Activity in LogicalCallContext, Activity [must be](https://msdn.microsoft.com/en-us/library/system.runtime.remoting.messaging.callcontext.logicalsetdata(v=vs.110).aspx) serializable 

It only affect the cross app-domain communication, which is not the case for Activity, however causes some implicit problems with MsTest

### CODE REVIEW

File: src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net45.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 namespace System.Diagnostics
 {
+    [Serializable]
```
Comment (stephentoub) [inline src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net45.cs]: Can you add a test for this?

And is there any value in making serializable everywhere, not just 4.5, if for no other reason than consistency?

**Comment by vancem** [comment]
It is not clear to me what actually happens when we try to serialize Activity.   My guess is that the linked list of tags and baggage will be lost.    If so arguably we would need to add 'GetDataObject' method (or maybe just marking the KeyValueListNode class Serializable would do it) .   

Do we know that things actually get fully serialized?  

I am a bit conflicted about this as arguably Serializable is either deprecated or at least frowned upon.   That argues for having #ifdefs for the platforms we need it for so that it is clear it is really not part of the contract going forward (unless we really believe we have to).  

Thus my inclination is to make this V4.5 only.

File: src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net45.cs (hunk 1 above)
Comment (lmolkova) [inline src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net45.cs]: This is for 4.5 only as declared in  Activity.Current.net45.cs (that is picked up for net45 target only).

**Comment by lmolkova** [comment]
@vancem 
You are right `KeyValueListNode` must also be serializable.  I have tested it and whole Activity is properly serialized.

Note that Serialized only appears in Activity.Current.net45.cs that makes it serializable on NET4.5 only.

However I'm concerned about allowing Activity to leave AppDomain. It is mutable and any changes done in one app domain after Start() will not be available in other domains.

**Comment by vancem** [comment]
I think this is OK.

**Pull request merged**